src/test/java/com/codurance/retropolis/controllers/CardControllerTest.java

```diff
@@ -89,38 +115,54 @@ public void returns_bad_request_when_column_is_not_found() throws Exception {
     given(cardService.addCard(any(NewCardRequestObject.class))).willThrow(new ColumnNotFoundException("Column Id is not valid"));
     NewCardRequestObject requestObject = new NewCardRequestObject("hello", 1L, "John Doe");
 
-    List<String> cardResponse = performHttpRequest(asJsonString(requestObject), status().isBadRequest());
+    List<String> cardResponse = performHttpPostRequest(asJsonString(requestObject), status().isBadRequest());
     assertEquals("Column Id is not valid", cardResponse.get(0));
   }
 
   @Test
   public void return_bad_request_when_text_is_empty() throws Exception {
     NewCardRequestObject requestObject = new NewCardRequestObject("", 1L, "John Doe");
     String content = asJsonString(requestObject);
-    List<String> errorResponse = performHttpRequest(content, status().isBadRequest());
+    List<String> errorResponse = performHttpPostRequest(content, status().isBadRequest());
     assertEquals("Text must not be less than 1 character", errorResponse.get(0));
   }
 
   @Test
   public void return_bad_request_when_columnId_is_null() throws Exception {
-    List<String> errorResponse = performHttpRequest("{\"text\":\"hello\",\"username\":\"John Doe\"}", status().isBadRequest());
+    List<String> errorResponse = performHttpPostRequest("{\"text\":\"hello\",\"username\":\"John Doe\"}",
+        status().isBadRequest());
     assertEquals("Column id cannot be empty", errorResponse.get(0));
   }
 
   @Test
   public void return_bad_request_when_no_text_is_sent() throws Exception {
-    List<String> errorResponse = performHttpRequest("{\"columnId\":\"1\",\"username\":\"John Doe\"}", status().isBadRequest());
+    List<String> errorResponse = performHttpPostRequest("{\"columnId\":\"1\",\"username\":\"John Doe\"}",
+        status().isBadRequest());
     assertEquals("Text cannot be empty", errorResponse.get(0));
   }
 
   @Test
   public void return_bad_request_when_no_username_is_sent() throws Exception {
-    List<String> errorResponse = performHttpRequest("{\"text\":\"hello\",\"columnId\":\"1\"}", status().isBadRequest());
+    List<String> errorResponse = performHttpPostRequest("{\"text\":\"hello\",\"columnId\":\"1\"}", status().isBadRequest());
     assertEquals("Username cannot be null", errorResponse.get(0));
   }
 
-  private <T> T performHttpRequest(String content, ResultMatcher response) throws Exception {
-    String responseBody = mockMvc.perform(MockMvcRequestBuilders.post(URL)
+  //TODO Refactor
+  private <T> T performHttpPostRequest(String content, ResultMatcher response) throws Exception {
+    MockHttpServletRequestBuilder post = MockMvcRequestBuilders.post(URL);
+    String responseBody = mockMvc.perform(post
+        .content(content)
+        .contentType(MediaType.APPLICATION_JSON)
+        .accept(MediaType.APPLICATION_JSON))
+        .andExpect(response).andReturn().getResponse().getContentAsString();
+
+    return objectMapper.readValue(responseBody, new TypeReference<>() {
+    });
+  }
```

Comment:
This method returns a generic response from the JSON in the test. It has helped to reduce duplication.